File: application/src/main/java/com/movie/application/service/MovieService.java
package com.movie.application.service;

import com.movie.application.dto.MovieResponseDto;
import com.movie.domain.dto.MovieSearchCondition;
import com.movie.domain.entity.Movie;
import com.movie.infra.repository.MovieJpaRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@RequiredArgsConstructor
@Transactional(readOnly = true)
public class MovieService {
    
    private final MovieJpaRepository movieRepository;
    
    public List<MovieResponseDto> getNowShowingMovies(MovieSearchCondition condition) {
        List<Movie> movies = movieRepository.findNowShowingMovies(condition);
        return movies.stream()
                .map(MovieResponseDto::from)
                .collect(Collectors.toList());
    }
} 

File: movies/src/main/kotlin/com/redis/movies/service/MovieService.kt
package com.redis.movies.service

import com.redis.movies.controller.dto.MovieResponse
import com.redis.movies.domain.Movie
import com.redis.movies.domain.MovieRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class MovieService(
    private val moviesRepository: MovieRepository
) {

    @Transactional(readOnly = true)
    fun gets(): MutableList<MovieResponse> {
        val movies = moviesRepository.findByReleaseDateOrderByDesc();
        return movies.stream().map { MovieResponse.fromEntity(it)}.toList()
    }
}